### PR TITLE
Implements newaxis

### LINKED ISSUE #24: Implement Newaxis
Newaxis is `None`.

### PR DESCRIPTION
**Pull request by melissawm**
Closes #24

### CODE REVIEW

File: ndindex/slice.py [hunk 1]
```diff
@@ -338,3 +338,55 @@ def reduce(self, shape=None, axis=0):
...
+        >>> from ndindex import Slice, Integer, Tuple
+        >>> idx = Slice(6)
+        >>> idx.newshape((8, 2))
+        (6, 2)
+        >>> idx.newshape(10)
+        (6,)
+        >>> idx.newshape((8, 10), axis=1)
+        (8, 6)
+        """
+        from . import Tuple
```
Comment (scopatz) [inline ndindex/slice.py:361]: Should be this at the top?

File: ndindex/slice.py [hunk 2]
```diff
@@ -338,3 +338,55 @@ def reduce(self, shape=None, axis=0):
...
+
+        if shape is None:
+            raise ValueError("shape can't be None")
+        elif isinstance(shape, Tuple):
+            raise TypeError("Tuple is not meant to be used as a shape - "
+                            "did you mean to use the built-in tuple type?")
+        else:
+            try:
+                shape = shape.raw
+            except:
```
Comment (scopatz) [inline ndindex/slice.py:373]: Should probably be `except Exception`

**Comment by scopatz** [comment]
Code looks good to me

File: ndindex/slice.py (hunk 1 above)
Comment (asmeurer) [reply ndindex/slice.py:361]: You can try it but it might fail due to circular imports.

File: ndindex/slice.py (hunk 2 above)
Comment (melissawm) [reply ndindex/slice.py:373]: Yeah, thanks!

**Comment by scopatz** [comment]
@asmeurer - please review and merge

File: ndindex/ndindex.py [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 from numpy import ndarray
 
+newaxis = None
```
Comment (asmeurer) [inline ndindex/ndindex.py:5]: This needs to be implemented as an NDIndex subclass.

Comment (melissawm) [reply ndindex/ndindex.py]: OK, I just went with that NumPy does. Let me know if this is what you had in mind

**Comment by asmeurer** [comment]
I've pushed updates here. This is ready for review.

**Comment by asmeurer** [comment]
Yes, getting the logic for newaxis right in the various Tuple methods was not easy. However, I've run the hypothesis test suite with lots of examples and no failures, so I'm fairly confident the code here is correct.

**Pull request merged**